### ISSUE TITLE
Not being able to "read" Manga

### ISSUE THREAD

**Issue by Shudhanshu-Raj** [lrorpilla/jidoujisho#166]

**Comment by Shudhanshu-Raj** [comment]
I tried 4 different mangas, none of them worked

**Comment by lrorpilla** [comment]
Hi @Shudhanshu-Raj, reading manga is a feature that has been deprecated in the Yuuna/2.x versions. I don't currently support 1.x anymore.

It's possible this feature might come back in the future in the form of a reader for Mokuro or Tachiyomi downloaded chapters but in the meantime, this project is in its hiatus post-2.4 and I will consider supporting reading manga again in the future once I've had a break.

**Comment by lrorpilla** [comment]
I can't. My plans are tentative and I don't have a roadmap or schedule as I develop the project in my free time. 

In the meantime I recommend using 1.x versions or using other community solutions to read manga.

**Comment by Shudhanshu-Raj** [comment]
I will try, thanks

**Comment by xrishox** [comment]
adding in web browser functionality should allow you to offload all of the work of supporting this to something like kavita. having a browser would also likely allow for support for any number of things that no one had even considered.  I'm not sure how feasible or difficult this would be though.

alternatively adding in pdf js alongside ttu would allow people to read manga using mokuro2pdf. 

https://github.com/mozilla/pdf.js

**Comment by Shudhanshu-Raj** [comment]
Thanks but I don't know how hard that might be, for the time being I figured out a way.

**Comment by lrorpilla** [comment]
I am working on this at the moment, and have recently experimented more with more native approaches to displaying Mokuro data. These were unsatisfactory so I am moving forward with a WebView-based approach. 

(Extra: For anyone interested who is developer-savvy there is a mokuro branch I am working on that has the partial progress on an attempt to render this in Flutter, but I decided not to go forward with it as the vertical text workarounds available for Flutter are extremely limited and disappointing.)

**Comment by xrishox** [comment]
I think that implementing pdf.js and using mokuro2pdf will both make it easier for you to implement and probably create a better user experience. mokuro's default ux is kinda not great and even less optimal on mobile whereas pdfs are far more intuitive and work better. The issue that I have run into is that as far as I'm aware there is nothing in the pdf spec that accounts for vertical text and most pdf tools don't properly support vertical text. pdf.js is one such implementation that does appear to properly support it. I'm not sure how difficult it would be to implement but I've got to imagine its easier than writing something from scratch.

**Comment by lrorpilla** [comment]
The [**2.5 preview**](https://github.com/lrorpilla/jidoujisho/releases/tag/2.5-preview) now supports embedding and mining on top of the Mokuro HTML overlay. While using the HTML overlay is a bit wonky, it still gives me a lot of control so I can make my own quality of life modifications.

I might consider supporting Mokuro2PDF in the future, but for now I believe this will suffice. One big factor in my decision is that it has to work with resources that are prominently available, I decided to support Mokuro HTMLs out of the box because they make up most of what are out there right now and I've found a lot more of them.

Will keep this thread open until 2.5 is out of preview.
